tests/infra_test.go

```diff
@@ -226,24 +226,26 @@ var _ = Describe("[Serial]Infrastructure", func() {
 			})
 
 			AfterEach(func() {
-				By("removing the taint from the tainted node")
-				var taints []k8sv1.Taint
+				if selectedNode != nil {
```

Comment:
I see; would it make sense to set the selected node in a `BeforeEach` clause then ?

Reply:
i dont think it make sense or needed, 
if the test reach the point it need it, it will set it, if it wasnt set it means the tests died before and didnt need it, 
and it would be simply not be cleaned, and no clean is needed, because no taints were added

Reply:
> i dont think its needed,
> if the test reach the point it need it, it will set it, if it wasnt set it means the tests died before and didnt need it,
> and it would be simply not be cleaned, and no clean is needed, because no taints were added

It is clearly not **needed**, I was merely pointing out it looks like this is something that should be in place before the test runs - e.g. this belongs in the test setup; the test itself should only check if XYZ is in the correct state.

> 
> its also not the purpose of the PR, and this code has a known open issue, so i prefer not to mix.

Understood.

> #4158